### ISSUE TITLE
"Cannot import url" error when working with standard json on newest mac version

### ISSUE THREAD

**Issue by shellygr** [ethereum/solidity#10059]
## Description

In Mac OS 10.15.7,  a single contract file compiled with standard-json with the input file provided as a URL, Solidity compiler fails to compile with the following error:
```
{"errors":[{"component":"general","formattedMessage":"Cannot import url (\"/Users/thomas/Code/MacSucks/Bank.sol\"): File outside of allowed directories.","message":"Cannot import url (\"/Users/thomas/Code/MacSucks/Bank.sol\"): File outside of allowed directories.","severity":"error","type":"IOError"}],"sources":{}}
```
A workaround is to provide a _relative_ path instead of the absolute path.
**This is a change of behavior compared to Ubuntu, Windows, and Mac OS 10.15.6.**

## Environment

- Compiler version: solc6.10
- Operating system: Mac OS 10.15.7

## Steps to Reproduce

This happens on any contract. But we take this solidity snippet:

```solidity
pragma solidity ^0.6.0;

contract Bank {
    struct Account {
        uint256 balance;
        uint256 exists;
    }
    mapping (address => Account) accounts;
    uint256 public totalSupply;

    function create(address a) public {
        require(accounts[a].exists == 0);
        accounts[a].exists = 1;
    }
}
```

This file is stored in `/Users/thomas/Code/MacSucks/Bank.sol`.

We prepare the following standard-json input:
```json
	{
		"language": "Solidity",
		"sources": {
			"/Users/thomas/Code/MacSucks/Bank.sol": {
				"urls": [
					"/Users/thomas/Code/MacSucks/Bank.sol"
				]
			}
		},
		"settings": {
			"remappings": [],
			"outputSelection": {
				"*": {
					"*": [
						"storageLayout",
						"abi",
						"evm.deployedBytecode",
						"evm.methodIdentifiers"
					],
					"": [
						"id",
						"ast"
					]
				}
			}
		}
	}
```

Here are a few attempts to run the compiler and what each one gives:
- `solc6.10 --allow-paths /Users/thomas/Code/MacSucks --standard-json < json.json` - fails on newest Mac, succeeds in other environments.
- `solc6.10 --standard-json < json.json` - fails on all environments
- `solc6.10 --allow-paths / --standard-json < json.json` - succeeds
- `solc6.10 --allow-paths . --standard-json < json.json` - succeeds
- `solc6.10 --allow-paths /Users/thomas/Code/MacSucks,. --standard-json < json.json` - succeeds

**Comment by cameel** [comment]
Thanks for the report! Could you check if this is still reproducible on MacOS when using the latest version of the compiler (0.7.3)? You can find static builds of all versions of the compiler in https://github.com/ethereum/solc-bin/tree/gh-pages/macosx-amd64.

Also, is `/Users/thomas/Code/MacSucks/Bank.sol` the original capitalization of that path on disk?

**Comment by shellygr** [comment]
I'm looping-in @thomasbernardi who has a Mac and with whose help I got this investigation going. 
I believe the capitalization presented is the correct one.

**Comment by cameel** [comment]
I suspect that this might be caused by symlinks. Is any part of the path `/Users/thomas/Code/MacSucks` a symbolic link by chance? `--allow-paths` by design requires you to specify the actual directory and does not work with symlinks.

**Comment by cameel** [comment]
By now I am pretty much convinced that this must have been caused by symlinks in workdir path and #11688 fixes that. So I'm going to close this once that PR is merged. Feel free to reopen if you encounter this even on a release including that fix (should be 0.8.8 or 0.8.9).
